clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerInterceptors.java

```
@@ -0,0 +1,90 @@
+
+
+import org.apache.kafka.clients.consumer.ConsumerInterceptor;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A container that holds the list {@link org.apache.kafka.clients.consumer.ConsumerInterceptor}
+ * and wraps calls to the chain of custom interceptors.
+ */
+public class ConsumerInterceptors<K, V> implements Closeable {
+    private final List<ConsumerInterceptor<K, V>> interceptors;
+
+    public ConsumerInterceptors(List<ConsumerInterceptor<K, V>> interceptors) {
+        this.interceptors = interceptors;
+    }
+
+    /**
+     * This is called when the records are about to be returned to the user.
+     * <p>
+     * This method calls {@link ConsumerInterceptor#onConsume(ConsumerRecords)} for each
+     * interceptor. Records returned from each interceptor get passed to onConsume() of the next interceptor
+     * in the chain of interceptors.
+     * <p>
+     * This method does not throw exceptions. If any of the interceptors in the chain throws an exception,
+     * it gets caught and ignored, and next interceptor in the chain is called with 'records' returned by the
+     * previous successful interceptor onConsume call.
+     *
+     * @param records records to be consumed by the client.
+     * @return records that are either modified by interceptors or same as records passed to this method.
+     */
+    public ConsumerRecords<K, V> onConsume(ConsumerRecords<K, V> records) {
+        ConsumerRecords<K, V> interceptRecords = records;
+        for (ConsumerInterceptor<K, V> interceptor: this.interceptors) {
+            try {
+                interceptRecords = interceptor.onConsume(interceptRecords);
+            } catch (Throwable t) {
+                // do not propagate interceptor exception, ignore and continue calling other interceptors
```

I was thinking that error level is too high, since, as Jason said, we are ignoring the errors. Ok, lets do it WARN level.
Good point about stacktrace,  I will fix.